src/CohortManager.js

```diff
@@ -0,0 +1,143 @@
+const Cohort = require('./Cohort.js')
+const Student = require('./Student.js')
+
+class CohortManager {
+  constructor() {
+    this.cohortList = []
+    this.studentId = 1
+  }
+
+  createCohort(name) {
+    this.#checkType(name, 'string', 'must search for a string')
+
+    const { ...newCohort } = new Cohort(name)
+    this.cohortList.push(newCohort)
+    return this.cohortList
+  }
+
+  searchForCohort(searchQuery) {
+    this.#checkType(searchQuery, 'string', 'must search for a string')
+
+    const searchResult = this.#findCohort(searchQuery.toLowerCase())
+
+    if (searchResult) {
+      return searchResult
+    }
+    throw new Error('no match found')
+  }
+
+  removeCohort(cohortName) {
+    this.#checkType(cohortName, 'string', 'must be a string')
+
+    const cohortToRemove = this.#findCohort(cohortName.toLowerCase())
+    this.#throwErrorIfUndefined(cohortToRemove)
+
+    const indexOfCohortToRemove = this.cohortList.indexOf(cohortToRemove)
+    this.cohortList.splice(indexOfCohortToRemove, 1)
+    return 'Removed successfully'
+  }
+
+  createStudent(firstName, lastName, gitHubUsername, email) {
+    const { ...newStudent } = new Student(
+      this.studentId,
+      firstName,
+      lastName,
+      gitHubUsername,
+      email
+    )
+    this.studentId++
+    return newStudent
+  }
+
+  addStudentToCohort(student, cohortName) {
+    this.#checkType(student, 'object', `student must be an object`)
+    this.#checkType(cohortName, 'string', `must be a string`)
+
+    const cohortToAddStudentTo = this.#findCohort(cohortName.toLowerCase())
+    this.#throwErrorIfUndefined(cohortToAddStudentTo)
+
+    const indexOfCohortToAddStudentTo =
+      this.cohortList.indexOf(cohortToAddStudentTo)
+    if (
+      this.cohortList[indexOfCohortToAddStudentTo].studentList.length === 24
+    ) {
+      throw new Error('cohort is full')
+    }
+    this.cohortList[indexOfCohortToAddStudentTo].studentList.push(student)
+    return `${student.firstName} added to ${cohortToAddStudentTo.name} successfully`
+  }
+
+  removeStudentFromCohort(student, cohortName) {
+    this.#checkType(student, 'object', `student must be an object`)
+    this.#checkType(cohortName, 'string', `must be a string`)
+
+    const cohortToRemoveStudentFrom = this.#findCohort(cohortName.toLowerCase())
+    this.#throwErrorIfUndefined(cohortToRemoveStudentFrom)
+
+    const indexOfCohortToRemoveStudentFrom = this.cohortList.indexOf(
+      cohortToRemoveStudentFrom
+    )
+
+    const studentToRemoveFromCohort = this.#findStudent(cohortName, student)
+    this.#throwErrorIfUndefined(studentToRemoveFromCohort)
+
+    const indexOfStudent = this.cohortList[
+      indexOfCohortToRemoveStudentFrom
+    ].studentList.indexOf(studentToRemoveFromCohort)
+
+    this.cohortList[indexOfCohortToRemoveStudentFrom].studentList.splice(
+      indexOfStudent,
+      1
+    )
```

Comment:
Hmm, I have to agree with Ed on this one. The `CohortManager` is modifying another classes state directly (even though you are destructuring it into a standard object elsewhere). You're kind of in a middle ground here now, so there are two paths to consider going down:

Option 1: Remove the use of classes for Students and Cohorts entirely (but keep the manager) and stick with Plain Old Javascript Objects

Option 2: Add `removeStudent` and `addStudent` methods to the `Cohort` class so it can be responsible for mutating its own state rather than letting somebody else stick their dirty fingers in its private space

Reply:
This code would be an ideal usage for somebody using your project:

```js
const cohort = cManager.findCohortByName('Cohort 7')
cohort.addStudent('Rick Sanchez')
```